tools/src/test/java/org/apache/kafka/tools/reassign/ReassignPartitionsIntegrationTest.java

```diff
@@ -0,0 +1,924 @@
+            brokerList = null;
+            Utils.closeQuietly(adminClient, "adminClient");
+            adminClient = null;
+            try {
+                TestUtils.shutdownServers(seq(servers), true);
+            } finally {
+                servers.clear();
+            }
+        }
+    }
+
+    private ReassignPartitionsCommand.VerifyAssignmentResult asScala(VerifyAssignmentResult res) {
+        Map<TopicPartition, ReassignPartitionsCommand.PartitionReassignmentState> partStates = new HashMap<>();
+        res.partStates.forEach((tp, state) -> partStates.put(tp, asScala(state)));
+
+        Map<TopicPartitionReplica, ReassignPartitionsCommand.LogDirMoveState> moveStates = new HashMap<>();
+        res.moveStates.forEach((tpr, state) -> moveStates.put(tpr, asScala(state)));
+
+        return new ReassignPartitionsCommand.VerifyAssignmentResult(asScala(partStates), res.partsOngoing, asScala(moveStates), res.movesOngoing);
+    }
+
+    @SuppressWarnings({"unchecked"})
+    private ReassignPartitionsCommand.PartitionReassignmentState asScala(PartitionReassignmentState state) {
+        return new ReassignPartitionsCommand.PartitionReassignmentState(
+            seq((List) state.currentReplicas),
+            seq((List) state.targetReplicas),
+            state.done
+        );
+    }
+
+    private ReassignPartitionsCommand.LogDirMoveState asScala(LogDirMoveState state) {
+        if (state instanceof ActiveMoveState) {
+            ActiveMoveState s = (ActiveMoveState) state;
+            return new ReassignPartitionsCommand.ActiveMoveState(s.currentLogDir, s.targetLogDir, s.futureLogDir);
+        } else if (state instanceof CancelledMoveState) {
+            CancelledMoveState s = (CancelledMoveState) state;
+            return new ReassignPartitionsCommand.CancelledMoveState(s.currentLogDir, s.targetLogDir);
+        } else if (state instanceof CompletedMoveState) {
+            CompletedMoveState s = (CompletedMoveState) state;
+            return new ReassignPartitionsCommand.CompletedMoveState(s.targetLogDir);
+        } else if (state instanceof MissingLogDirMoveState) {
+            MissingLogDirMoveState s = (MissingLogDirMoveState) state;
+            return new ReassignPartitionsCommand.MissingLogDirMoveState(s.targetLogDir);
+        } else if (state instanceof MissingReplicaMoveState) {
+            MissingReplicaMoveState s = (MissingReplicaMoveState) state;
+            return new ReassignPartitionsCommand.MissingReplicaMoveState(s.targetLogDir);
+        }
+
+        throw new IllegalArgumentException("Unknown state " + state);
+    }
+
+    @SuppressWarnings("unchecked")
+    static <T> scala.collection.immutable.Set<T> set(final T... set) {
+        return mutableSet(set).toSet();
+    }
+
+    @SuppressWarnings({"deprecation", "unchecked"})
+    private static <T> scala.collection.mutable.Set<T> mutableSet(final T...set) {
+        return JavaConverters.asScalaSet(new HashSet<>(Arrays.asList(set)));
```

Comment:
> Is this no longer the case?

I spend some time to find one during implementation of #14355 , actually. 
All methods I found either don't exists in scala 2.12 or deprecated in scala 2.13.

Do you know the way to convert collections scala <-> java that will work both in scala 2.12 and scala 2.13 and will not be deprecated in scala 2.13?

Anyway, these methods will be removed in #13247 .

Reply:
Hmm. I think I saw some examples of `scala.collection.JavaConverters` that didn't have deprecation warnings on them. I did a search of `asScala` in the java files and looked if the files had deprecation warnings. There were a few that seemed to not.

If we remove these, then maybe it's ok. We should file a JIRA to clean them up though.

Reply:
> We should file a JIRA to clean them up though

Jira not required - because in #13247 I will remove `ReassignPartitionCommand.scala` therefore all POJO inside it.
So all tests will forefully user java version of clasees.

The goal of this PR is to simplify review of the single test conversion and reduce changes in #13247